### PR TITLE
fix(Radio Garden): update for new element classes

### PR DESCRIPTION
**Pull request by awooooool**
**Describe the changes in this pull request:**
Found out that the presence doesn't work anymore and tons of error in console, so I'm trying to make it work again.

<details>
<summary> Proof showing the creation/modification is working as expected </summary>

Screenshot 2022-02-06 170858

Screenshot 2022-02-06 170923

Screenshot 2022-02-06 171000

Screenshot 2022-02-06 171035

Screenshot 2022-02-06 171105

Screenshot 2022-02-06 171120

</details>

### CODE REVIEW

File: websites/R/Radio Garden/presence.ts [hunk 1]
```diff
@@ -2,42 +2,45 @@ const presence = new Presence({
...
 
-let details: HTMLInputElement, state: HTMLInputElement;
+let details: HTMLInputElement, state: HTMLInputElement, currentState: string;
 
 presence.on("UpdateData", async () => {
 	if (!document.title.includes("Explore live radio by rotating")) {
-		details = document.querySelector(".ChannelTitle_title__2QQj5");
-		state = document.querySelector(".ChannelTitle_subtitle__DZ_ZQ");
+		details = document.querySelector("._title_165lg_36"); // which radio we're tuning right now
+		state = document.querySelector("._subtitle_165lg_43"); // the state where the radio serves
```
Comment (N0chteil) [inline websites/R/Radio Garden/presence.ts:10]:
```suggestion
		details = document.querySelector("._title_165lg_36")?.textContent ?? "Unknown Radio"; // which radio we're tuning right now
		state = document.querySelector("._subtitle_165lg_43")?.textContent ?? "Unknown Region/Country"; // the state where the radio serves
```

File: websites/R/Radio Garden/presence.ts [hunk 2]
```diff
@@ -2,42 +2,45 @@ const presence = new Presence({
...
+		currentState = document
+			.getElementsByClassName("_control_oyndo_11 _modPlay_oyndo_53")[0]
+			.getAttribute("aria-label"); // get current state of the "button", not the "player", except when loading
+		if (currentState.length > 0) {
+			if (currentState === "stop") {
+				// currently playing
+				presenceData.smallImageKey = "statusplay";
+				presenceData.smallImageText = "Playing";
+				presenceData.details = details.textContent;
+				presenceData.state = state.textContent;
```
Comment (N0chteil) [inline websites/R/Radio Garden/presence.ts:28]:
```suggestion
				presenceData.details = details;
				presenceData.state = state;
```

File: websites/R/Radio Garden/presence.ts [hunk 3]
```diff
@@ -2,42 +2,45 @@ const presence = new Presence({
...
 	if (!document.title.includes("Explore live radio by rotating")) {
-		details = document.querySelector(".ChannelTitle_title__2QQj5");
-		state = document.querySelector(".ChannelTitle_subtitle__DZ_ZQ");
+		details = document.querySelector("._title_165lg_36"); // which radio we're tuning right now
+		state = document.querySelector("._subtitle_165lg_43"); // the state where the radio serves
+
 		const elapsed = Math.floor(Date.now() / 1000),
 			presenceData: PresenceData = {
 				details: details.textContent,
 				state: state.textContent,
```
Comment (N0chteil) [inline websites/R/Radio Garden/presence.ts:15]:
```suggestion
				details: details,
				state: state,
```

**Review by 7loud — COMMENTED** [review]
Some suggestions

File: websites/R/Radio Garden/presence.ts (hunk 1 above)
Comment (awooooool) [reply websites/R/Radio Garden/presence.ts:10]: How about this? From what I could see, the only way the element isn't there is when the site is currently loading.
<details>
<summary>Screenshot of the loading page</summary>
<br>

Screenshot 2022-02-07 060441

</details>

```suggestion
		details = document.querySelector("._title_165lg_36")?.textContent ?? "Loading..."; // which radio we're tuning right now
		state = document.querySelector("._subtitle_165lg_43")?.textContent ?? "Loading..."; // the state where the radio serves
```

Comment (N0chteil) [reply websites/R/Radio Garden/presence.ts:10]: I think you can understand it better that way:
```suggestion
		details = document.querySelector("._title_165lg_36")?.textContent ?? "Loading Radio..."; // which radio we're tuning right now
		state = document.querySelector("._subtitle_165lg_43")?.textContent ?? "Loading Region/Country..."; // the state where the radio serves
```

Comment (awooooool) [reply websites/R/Radio Garden/presence.ts]: I just realized the `if` just below `.on('UpdateData')` actually checks if the site still loading or not. So I reverted back to your first suggestion. Do please take a look at the commit that I just pushed.

Thanks in advance.

Comment (N0chteil) [reply websites/R/Radio Garden/presence.ts]: 👍

**Review by Bas950 — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**